go-controller/pkg/ovn/controller/services/service_tracker.go

```diff
@@ -18,6 +18,13 @@ func virtualIPKey(virtualIP string, protocol v1.Protocol) string {
 	return virtualIP + "/" + string(protocol)
 }
 
+// virtualIPKeyWithPort returns a string used for the virtual IPs index.
+// it accepts an ip / port pair
+func virtualIPKeyWithPort(virtualIP string, port int32, protocol v1.Protocol) string {
```

Comment:
Not sure. All the *key functions are here, this is just another way to generate a key for the service tracker.

```
// virtualIPKey returns a string used for the virtual IPs index.
func virtualIPKey(virtualIP string, protocol v1.Protocol) string {
	return virtualIP + "/" + string(protocol)
}

// virtualIPKeyWithPort returns a string used for the virtual IPs index.
// it accepts an ip / port pair
func virtualIPKeyWithPort(virtualIP string, port int32, protocol v1.Protocol) string {
	vip := util.JoinHostPortInt32(virtualIP, port)
	return virtualIPKey(vip, protocol)
}

// splitVirtualIPKey splits the VirtualIPKey from the service tracker in virtual ip and protocol
func splitVirtualIPKey(key string) (string, v1.Protocol) {
	parts := strings.Split(key, "/")
	return parts[0], v1.Protocol(parts[1])
}
```

Said that, I can move it if you want.

Reply:
After our discussion I finally realize that the other VirtualIPKey is being used with port too, its just port is overloaded into the virtualIP parameter. Rather than add confusion with multiple paths here. Let's remove the virtualIPKeyWithPort and just use the virtualIPKey function. It makes it consistent, although less optimal. Then let's add doctext on func virtualIPKey to say // virtualIP is a combination of IP and port

We can fix the bad code later where we always do a vip join before we call the function.